packages/terra-list/src/List.jsx

```diff
@@ -150,7 +173,53 @@ const List = ({
     attrSpread['aria-label'] = intl.formatMessage({ id: 'Terra.list.multiSelect' });
   }
 
-  return (
+  const reorderListItems = (list, startIndex, endIndex) => {
+    const result = Array.from(list);
+    const [removed] = result.splice(startIndex, 1);
+    result.splice(endIndex, 0, removed);
+    return result;
+  };
+
+  const handleDragEnd = (result, provided) => {
+    const listItems = listNode && listNode.querySelectorAll('[data-item-show-focus]');
+    // dropped outside the list
+    if (!result.destination) {
+      provided.announce(intl.formatMessage({ id: 'Terra.list.cancelDrag' }, { startPosition: (result.source.index + 1) }));
+      return;
+    }
+    const items = reorderListItems(
+      listItem,
+      result.source.index,
+      result.destination.index,
+    );
+    setlistItem(items);
+    if (listItems && listItems[result.source.index]) {
+      listItems[result.source.index].focus();
+    }
+    provided.announce(intl.formatMessage({ id: 'Terra.list.drop' }, { startPosition: (result.source.index + 1), endPosition: (result.destination.index + 1) }));
+    if (onDragEnd) {
+      onDragEnd(result, provided);
+    }
+  };
+
+  const handleDragStart = (start, provided) => {
+    provided.announce(intl.formatMessage({ id: 'Terra.list.lift' }, { startPosition: (start.source.index + 1) }));
+  };
+
+  const handleDragUpdate = (update, provided) => {
+    if (update.destination) {
+      provided.announce(intl.formatMessage({ id: 'Terra.list.drag' }, { startPosition: (update.source.index + 1), endPosition: (update.destination.index + 1) }));
+    }
+  };
+
+  const cloneListItem = (ListItem, provider) => React.cloneElement(ListItem, {
```

Comment:
Do we need to use `cloneElement` here? It seems like the only place this is used is at `renderDraggableListDom` function. Couldn't we just specify the component directly and spread the necessary props there?

Reply:
No. Just adding draggable props to the element. where component is invoked in the top level and we are receiving as a children object. Thanks